### PR TITLE
LC-003 Migrated to new repository

### PR DESCRIPTION
**Pull request by daviddl**

### CODE REVIEW

**Comment by daviddl** [comment]
@richardlalancette  please review

File: src/quest/QuestParser.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#include "QuestParser.h"
+#include <utility\YiRapidJSONUtility.h>
+
+QuestModel* QuestParser::ParseQuest(const CYIString &filepath)
```
Comment (rlalance) [inline src/quest/QuestParser.cpp:4]: Move to QuestModel and rename to FromJson.
Move the Objective parsing into QuestObjective and add FromJson.

File: src/quest/QuestObjectiveModel.cpp [hunk 2]
```diff
@@ -3,18 +3,81 @@
...
+
+void QuestObjectiveModel::SetUnresolvedText(CYIString text)
+{
+    if (GetRowCount() == 0)
+    {
+        InsertRow(0);
+    }
+
+    SetResolutionText(0, text);
+    m_ncurrentState = 0;
```
Comment (rlalance) [inline src/quest/QuestObjectiveModel.cpp:30]: m_nCurrentState.
What is this for?

File: src/quest/QuestObjectiveModel.cpp [hunk 3]
```diff
@@ -3,18 +3,81 @@
...
+void QuestObjectiveModel::AddResolutionText(CYIString text)
+{
+    YI_INT32 count = GetRowCount();
+    YI_INT32 columns = GetColumnCount();
+    InsertRow(count);
+    count = GetRowCount();
+    CYIDataModelIndex index = GetIndex(count - 1, 0);
+    CYIAny resolutionText = CYIAny(text);
+    
+    if (index.IsValid()) // Comes out invalid despite being within the bounds of the datamodel
```
Comment (rlalance) [inline src/quest/QuestObjectiveModel.cpp:42]: Is this still true?

File: src/quest/QuestObjectiveModel.cpp [hunk 4]
```diff
@@ -3,18 +3,81 @@
...
+            objectiveInfo.Append(resolution);
+            
+            if (i < GetRowCount()-1)
+            {
+                objectiveInfo.Append(" ");
+            }
+        }
+    }
+
+    objectiveInfo.Append(CYIString("]"));
```
Comment (rlalance) [inline src/quest/QuestObjectiveModel.cpp:73]: space before return statement.

File: src/quest/QuestModel.cpp [hunk 5]
```diff
@@ -1,61 +1,95 @@
 #include <utility\YiString.h>
```
Comment (rlalance) [inline src/quest/QuestModel.cpp:1]: Application includes before SDK includes.
And please sort them alphabetically.

File: src/quest/QuestParser.cpp [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    CYIRapidJSONUtility::GetStringField(&quest, "Name", questName, parsingError);
+    YI_ASSERT(!parsingError.HasError(), "QuestParser", parsingError.GetParsingErrorMessage());
+
+    CYIRapidJSONUtility::GetStringField(&quest, "Description", questDescription, parsingError);
+    YI_ASSERT(!parsingError.HasError(), "QuestParser", parsingError.GetParsingErrorMessage());
+
+    QuestModel* newQuest = new QuestModel(questName, questDescription);
+
+    //Parse Objectives --------------------
+    const yi::rapidjson::Value& objectives = quest["Objectives"];
```
Comment (rlalance) [inline src/quest/QuestParser.cpp:28]: So this chunk can go in QuestObjectiveModel

File: src/quest/QuestParser.h [hunk 7]
```diff
@@ -1,59 +1,11 @@
-#ifndef QUEST_PARSER
-#define QUEST_PARSER
+#ifndef QUEST_PARSER_H
```
Comment (rlalance) [inline src/quest/QuestParser.h:1]: You should be able to get rid of Questparser.h.

File: src/quest/QuestModel.cpp [hunk 8]
```diff
@@ -1,61 +1,95 @@
...
-	CYIString result;
-	result.Append(name);
-	result.Append(CYIString("\n"));
-	result.Append(description);
+    CYIString questInfo;
+    questInfo.Append(m_name);
+    questInfo.Append(CYIString("\n"));
+    questInfo.Append(m_description);
+
+//    YI_INT32 count = GetRowCount();
```
Comment (rlalance) [inline src/quest/QuestModel.cpp:28]: remove comment.

File: src/quest/QuestModel.cpp [hunk 9]
```diff
@@ -1,61 +1,95 @@
...
 
-	return result;
+    for (YI_INT32 i = 1; i < GetRowCount(); i++) //Skipping the first one because it is empty. Design flaw.
```
Comment (rlalance) [inline src/quest/QuestModel.cpp:30]: Remove comment if the DataModel work as designed.

File: src/quest/QuestModel.cpp [hunk 10]
```diff
@@ -1,61 +1,95 @@
...
+    return questInfo;
 }
 
 void QuestModel::AddObjective(CYIString name, const std::vector<CYIString> resolutions)
 {
-	InsertRow(GetRowCount() - 1);
-	
-	
-}
+    InsertRow(GetRowCount());
```
Comment (rlalance) [inline src/quest/QuestModel.cpp:48]: you don't need to address this now, but maybe we can create our own DataModel class that would have a method call InsertLast? later....

File: src/quest/QuestModel.cpp [hunk 11]
```diff
@@ -1,61 +1,95 @@
...
-	for (YI_INT32 i = 0; i < nItems; i++)
-	{
-		CYIAny listData = model2->GetItemData(model2->GetIndex(i, 0));
-		YI_LOGD("DATAMODEL", "Contains %s at index %d", listData.Get<CYIString>().GetData(), i);
-	}
-
-	free(model2);
+    //Populate
+    YI_INT32 nItems = 10;
+    CYIAbstractDataModel* model2 = new CYIAbstractDataModel(nItems, 1); //Now I know that this can be instantiated, even though I have yet to see an implementation of the class.
```
Comment (rlalance) [inline src/quest/QuestModel.cpp:79]: model2 needs a more descriptive name.

File: src/quest/QuestModel.cpp [hunk 12]
```diff
@@ -1,61 +1,95 @@
...
+
+    CYIAbstractDataModel model(nItems, 1); // nItems rows, 1 column
```
Comment (rlalance) [inline src/quest/QuestModel.cpp:81]: Better name too.

File: src/quest/QuestModel.cpp (hunk 11 above)
Comment (rlalance) [inline src/quest/QuestModel.cpp:79]: remove comments.

**Comment by rlalance** [comment]
Add end of file newlines to all required files.

**Pull request merged**